code/modules/power/singularity/generator.dm

```diff
@@ -13,7 +13,8 @@
 /obj/machinery/the_singularitygen/process()
 	var/turf/T = get_turf(src)
 	if(src.energy >= 200)
-		new creation_type(T, 50)
+		var/obj/singularity/S = new creation_type(T, 50)

```

Comment:
You don't have to typecast a singularity here, especially since one of these also makes the Tesla Ball.